knpy/braid_vec.py

```diff
@@ -0,0 +1,371 @@
+# pylint: disable=R0801
+from typing import Callable
+import numpy as np
+import torch
+import braidvisualiser as bv
+from functools import partial, wraps
+from .data_utils import knots_in_braid_notation_dict
+from .exceptions import IllegalTransformationException, InvalidBraidException, IndexOutOfRangeException
+
+from . import braid_cpp_impl as B
+
+type BraidNotation = np.ndarray
+type BraidTransformation = Callable[[], "Braid"]
+
+
+def braid_move(fun: Callable):
+    """
+    Decorator that transforms C++ exceptions into ``IllegalTransformationException``
+    """
+
+    @wraps(fun)
+    def wrapped(*args, **kwargs):
+        try:
+            return fun(*args, **kwargs)
+        except B.IllegalTransformationException as e:
+            raise IllegalTransformationException(e) from e
+
+    return wrapped
+
+
+class Braid:
+
+    def __init__(self, sigmas: np.ndarray | list[int] | str, notation_index: int = 0, copy_sigmas: bool = True):
+        """
+        Init Braid class, sigmas should not contain zero or bigger value than n_strands
+
+        sigmas: Braid notation, e.g. [1,-1,2] or the string name of knot e.g. 4_1 #TODO Above 10 there a and n knots
+            (e.g. 11a,13n)
+        notation_index: If sigmas is a name of braid than it is possible that multiple notations are available to the
+            same knot. notation_index says which one to choose from these.
+        #TODO 10_136 {{-1;-1;-2;3;-2;1;-2;-2;3;2;2};{-1;2;-1;2;3;-2;-2;-4;3;-4}}? Which one?
+        #TODO 11n_8,{{-1;-1;-2;1;-2;-1;3;-2;-2;-4;3;-4};{1;2;-1;2;3;-2;-1;-1;-2;-2;-3;-3;-2}}? Which one?
+        """
+        self._braid: BraidNotation
+        if isinstance(sigmas, str):
+            self._braid = np.array(knots_in_braid_notation_dict[sigmas][notation_index])
```

Comment:
`dtype` should be set to `np.int32`, same as in `braid.py`.

Reply:
Done